Source/WebKit/GPUProcess/graphics/RemoteGraphicsContextGL.cpp

```diff
@@ -232,7 +234,9 @@ void RemoteGraphicsContextGL::paintCompositedResultsToCanvas(WebCore::RenderingR
 void RemoteGraphicsContextGL::paintCompositedResultsToCanvasWithQualifiedIdentifier(QualifiedRenderingResourceIdentifier imageBuffer, CompletionHandler<void()>&& completionHandler)
 {
     assertIsCurrent(workQueue());
-    paintPixelBufferToImageBuffer(m_context->readCompositedResultsForPainting(), imageBuffer, WTFMove(completionHandler));
+    m_context->withDisplayBufferAsPlatformImage([&](PlatformImagePtr&& image) {
```

Comment:
```
// We know that the image might be updated afterwards, so flush the drawing so that read back does not
occur.
imageBuffer->flushDrawingContext();
```
(this is not so self-evident at the moment for the composited image, but it's a good starting point anyway.